SU2_CFD/include/gradients/computeGradientsGreenGauss.hpp

```diff
@@ -153,29 +233,100 @@ void computeGradientsGreenGauss(CSolver* solver,
         if (!nodes->GetDomain(iPoint)) continue;
 
         su2double volume = nodes->GetVolume(iPoint) + nodes->GetPeriodicVolume(iPoint);
-
         const auto area = geometry.vertex[iMarker][iVertex]->GetNormal();
 
-        for (size_t iVar = varBegin; iVar < varEnd; iVar++)
-        {
-          su2double flux = field(iPoint,iVar) / volume;
+        // When the node is shared with a symmetry we need to mirror the contribution of
+        // the face that is coincident with the inlet/outlet
+        if (nodes->GetSymmetry(iPoint) && nodes->Getinoutfar(iPoint)) {
+
+          // we have to find the edges that were missing in the symmetry computations.
+          // So we find the jPoints that are on the inlet plane
+          // so we loop over all neighbor of iPoint, find all jPoints and then check if it is on the inlet
+          unsigned long jPoint = 0;
+          for (size_t iNeigh = 0; iNeigh < nodes->GetnPoint(iPoint); ++iNeigh) {
+            jPoint = nodes->GetPoint(iPoint, iNeigh);
+            if (nodes->Getinoutfar(jPoint)) {
+              cout << "  jPoint " << jPoint << " is on the inlet plane" << endl;
+              break;
+            }
+          }
+
+          su2double dir = (iPoint < jPoint) ? 1.0 : -1.0;
+          su2double weight = dir / (2.0*volume);
+
+          // this edge jPoint - jPoint is the missing edge for the symmetry computations
+          //compute the flux on the face between iPoint and jPoint
+          for (size_t iVar = varBegin; iVar < varEnd; ++iVar) {
+            // average of i and j at the midway point
+            flux[iVar] = weight * 0.5 * (field(iPoint, iVar) + field(jPoint, iVar));
+            /* original : flux[iVar] = field(iPoint,iVar) / volume; */
+            fluxReflected[iVar] = flux[iVar];
+            cout << "flux ("<<iPoint<<","<<iVar<<")="<<flux[iVar]<<endl;
+          }
+
+          // find the point iPoint on the symmetry plane and get the vertex normal at ipoint wrt the symmetry plane
+          const su2double* VertexNormal = getVertexNormalfromPoint(config, geometry,iPoint);
+          cout << "  vertex normal = " << VertexNormal[0] <<", " << VertexNormal[1] << endl;
+          // get the normal on the vertex
+
+          // now reflect in the mirror
+          // reflected normal V=U - 2U_t
+          const auto NormArea = GeometryToolbox::Norm(nDim, VertexNormal);
```

Comment:
My understanding of the book is that we take out normal components, reflection should not be necessary